sunpy/net/dataretriever/sources/goes.py

```diff
@@ -118,43 +118,26 @@ def _get_url_for_timerange(self, timerange, **kwargs):
 
         Parameters
         ----------
-        timerange: sunpy.time.TimeRange
-            time range for which data is to be downloaded.
-        satellitenumber : int
-            GOES satellite number (default = 15)
-        data_type : str
-            Data type to return for the particular GOES satellite. Supported
-            types depend on the satellite number specified. (default = xrs_2s)
+        timerange : `~sunpy.time.TimeRange`
+
+        Returns
+        -------
+        list :
+            The URL(s) for the corresponding timerange.
         """
-        # find out which satellite and datatype to query from the query times
-        base_url = "https://umbra.nascom.nasa.gov/goes/fits/"
-        start_time = Time(timerange.start.strftime("%Y-%m-%d"))
-        # make sure we are counting a day even if only a part of it is in the query range.
-        day_range = TimeRange(
-            timerange.start.strftime("%Y-%m-%d"), timerange.end.strftime("%Y-%m-%d")
-        )
-        total_days = int(day_range.days.value) + 1
-        result = list()
-
-        # Iterate over each day in the input timerange and generate a URL for
-        # it.
-        for day in range(total_days):
-            # It is okay to convert to datetime here as the start_time is a date
-            # hence we don't necesserily gain anything.
-            # This is necessary because when adding a day to a Time, we may
-            # end up with the same day if the day is a leap second day
-            date = start_time.datetime + timedelta(days=day)
-            regex = date.strftime("%Y") + "/go{sat:02d}"
-            if date < parse_time("1999/01/15"):
-                regex += date.strftime("%y%m%d") + ".fits"
-            else:
-                regex += date.strftime("%Y%m%d") + ".fits"
-            satellitenumber = kwargs.get(
-                "satellitenumber", self._get_goes_sat_num(date)
-            )
-            url = base_url + regex.format(sat=satellitenumber)
-            result.append(url)
-        return result
+        timerange = TimeRange(timerange.start.strftime('%Y-%m-%d'), timerange.end)
+        if timerange.start < parse_time("1999/01/15"):
```

Comment:
You check only `timerange.start` here, so it's possible that `timerange.end` will be after this date, so you may need to split the time range in two.

Reply:
I've added functionality now to take this into account 👍